test/libdnf/rpm/test_package.cpp

```diff
@@ -0,0 +1,346 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "test_package.hpp"
+
+#include "../utils.hpp"
+
+#include "libdnf/rpm/solv_query.hpp"
+
+#include <vector>
+
+
+CPPUNIT_TEST_SUITE_REGISTRATION(RpmPackageTest);
+
+
+libdnf::rpm::Package RpmPackageTest::get_pkg(const std::string &nevra) {
+    libdnf::rpm::SolvQuery query(sack.get());
+    query.ifilter_nevra(libdnf::sack::QueryCmp::EQ, {nevra});
+    CPPUNIT_ASSERT_MESSAGE("get_pkg(\"" + nevra + "\"): no package found.", query.size() > 0);
+    return *query.get_package_set().begin();
+}
+
+
+void RpmPackageTest::test_equality() {
+    auto package1 = get_pkg("CQRlib-1.1.1-4.fc29.x86_64");
+    auto package2 = get_pkg("CQRlib-1.1.1-4.fc29.x86_64");
+    CPPUNIT_ASSERT(package1 == package2);
```

Comment:
Is it possible tu use `CPPUNIT_ASSERT_EQUAL`?

Reply:
Not here, `CPPUNIT_ASSERT_EQUAL` doesn't use `operator==`, it's got its own set of classes for this (you can see the commit for asserting std collections for a rough idea, I didn't dig much further). Also, I am testing the `operator==` itself, so that's what I should be calling.

Thinking about it now, I should also test `operator==` for when the packages aren't equal and `operator!=` when they are equal...